### ISSUE TITLE
No writing on Boox Note Air 3C?

### ISSUE THREAD

**Issue by Qiqayon** [olup/notable#34]
This app looks wonderful and I would love to try and use it. 
It installs fine on my Boox Note Air 3C (V3.5 firmware). 
The screen comes up, pens can be selected, tapping works...
but nothing happens when I use the pen i.e. no writing is visible.

**Comment by zuhairzah** [comment]
Same as mine, the app does not work on Note Air 3C

**Comment by fredcl** [comment]
same for me on my Note Air 3C, maybe white color of the fonts ? (but I can't select a font color....)

**Comment by scientificsound** [comment]
just got my Note Air 3 (b&w version) and i'm having the same problem.  shame b/c it looks like a great app...

**Comment by wvxx** [comment]
+1. Would love to see this functioning properly on my Note Air 3C.

**Comment by aarontharris** [comment]
Same for Note Air 3 (B&W)
My guess is the Onyx SDK needs to be updated.

    implementation('com.onyx.android.sdk:onyxsdk-device:1.2.26') // 1.2.27 is latest
    implementation('com.onyx.android.sdk:onyxsdk-pen:1.4.8') // up to date
    implementation('com.onyx.android.sdk:onyxsdk-base:1.6.42') // 1.6.51 is latest

I will check it out and try updating these and report back :)

**Comment by aarontharris** [comment]
> Same for Note Air 3 (B&W) My guess is the Onyx SDK needs to be updated.
> 
> ```
> implementation('com.onyx.android.sdk:onyxsdk-device:1.2.26') // 1.2.27 is latest
> implementation('com.onyx.android.sdk:onyxsdk-pen:1.4.8') // up to date
> implementation('com.onyx.android.sdk:onyxsdk-base:1.6.42') // 1.6.51 is latest
> ```
> 
> I will check it out and try updating these and report back :)

Unfortunately this did work.

Microsoft OneNote implements the Onyx SDK and it's pen functions are smooth. This suggests the maven repo is not up to date with the latest SDK and Microsoft is using some other source, possibly github/Onyx-Intl.

**Comment by aarontharris** [comment]
Ok I fixed it.  I will submit a pull request.

**Comment by aarontharris** [comment]
OK Pull request submitted [pull/36](https://github.com/olup/notable/pull/36)

TLDR;
I checked out the onyx demo and discovered it was working on my Note Air 3. How could that be?
So I started stripping unnecessary things from their app until it stopped working.

What I found was:
App.onCreate() needs to initialize their RxManager & HiddenApiBypass.
I assume SDK30 caused issues with the Onyx SDK and they implemented the HiddenApiBypass as a workaround.
It seems to have fixed my local copy of notable app for my Note Air 3 (B&W) as well as my own Onyx app (work in progress).

Hopefully if @olup pulls it, everyone should get the fix :)

Thanks for the app!

**Comment by olup** [comment]
Hey ! This is good news, do you want to try a quick pr for this ? Do you reckon it will also work on following color devices ?

**Comment by aarontharris** [comment]
> Hey ! This is good news, do you want to try a quick pr for this ? Do you reckon it will also work on following color devices ?

There is a PR already submitted :)

I suspect the issue has to do with Android SDK30 rather than Note Air 3 or 3C. So my feeling is that this will fix the issue for color devices as well. 🤞

**Comment by aarontharris** [comment]
Looks like [pull/36](https://github.com/olup/notable/pull/36) was merged. This issue should be closed.

Thanks @olup

**Comment by olup** [comment]
Thanks to you !

**Comment by aarontharris** [comment]
@olup I almost forgot -- I do not have the ability to test on a Note Air 2. So someone should probably test that and see if this change is backwards compatible (or if we need an abstraction per device).

**Comment by wvxx** [comment]
I'm happy to confirm it now works flawlessly on my Note Air 3C. Thank you @aarontharris @olup. Well done!

**Comment by 50meguy** [comment]
great job on the app.  I got the updated version to work on an air 3 b&w; whatever SW firmware version latest is.    Unfortunately, the new beta just crashes after some use....typically when using a pencil and just drawing doodles.   Am I the only one?
